src/components/Pagination/Pagination.tsx

```diff
@@ -55,21 +58,24 @@ export function Pagination({
   previousKeys,
   plain,
   accessibilityLabel,
+  label,
 }: PaginationProps) {
   const intl = useI18n();
 
   const node: React.RefObject<HTMLElement> = React.createRef();
-  let label: string;
 
-  if (accessibilityLabel) {
-    label = accessibilityLabel;
-  } else {
-    label = intl.translate('Polaris.Pagination.pagination');
-  }
+  const navLabel =
+    accessibilityLabel || intl.translate('Polaris.Pagination.pagination');
 
   const className = classNames(styles.Pagination, plain && styles.plain);
-  const previousClassName = classNames(styles.Button, styles.PreviousButton);
-  const nextClassName = classNames(styles.Button, styles.NextButton);
+
+  const previousClassName = !label
+    ? classNames(styles.Button, styles.PreviousButton)
+    : classNames(styles.Button);
+
+  const nextClassName = !label
+    ? classNames(styles.Button, styles.NextButton)
+    : classNames(styles.Button);
```

Comment:
This duplication would make adding new classes to buttons awkward. I think it might be simpler to put the condition on the class we're targetting with our condition instead.

```js
const nextClassName = classNames(
  styles.Button,
  !label && styles.NextButton,
);
```

Reply:
Good catch, this is a better way of doing it 💯